libs/librepcb/core/project/board/drc/boarddesignrulecheckdata.cpp

```diff
@@ -86,14 +87,25 @@ BoardDesignRuleCheckData::BoardDesignRuleCheckData(
                               nl->getEndPoint().getPosition(), nl->getWidth(),
                               &nl->getLayer()});
     }
-    foreach (const BI_Via* via, ns->getVias()) {
+    foreach (const BI_Via* bi_via, ns->getVias()) {
+      QSet<const Layer*> connectedLayers;
+
+      // Add all the layers of traces directly connected to the current via to
+      // the connectedLayers set. The via may be connected to more layers
+      // through other mechanisms like planes, but identifying those connections
+      // can be expensive, so it's not done here.
+      foreach (const BI_NetLine* nl, bi_via->getNetLines()) {
+        connectedLayers.insert(&nl->getLayer());
+      }
+
+      const librepcb::Via via = bi_via->getVia();
```

Comment:
I wanted to ignore that for now but as another change needs to be made anyway, this could be improved a bit too (avoid unnecessary copy):

```suggestion
      const librepcb::Via& via = bi_via->getVia();
```

Btw actually we never use snake_case variable names... I this case I simply wouldn't create the `via` variable to avoid the need for renaming `bi_via` but it's clearly a matter of taste so I'll accept your way :slightly_smiling_face: No need to change anything here, only the `&` would be nice.